site/zenodo_rdm/legacy/generators.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Zenodo is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Zenodo legacy permissions generators."""
+
+from invenio_rdm_records.services.generators import IfRestricted
+from invenio_records.dictutils import dict_lookup
+
+
+class IfFilesRestrictedForCommunity(IfRestricted):
+    """Conditional generator for files restriction for community."""
+
+    def __init__(self, then_, else_):
+        """Constructor."""
+        super().__init__("files", then_, else_)
+
+    def _condition(self, record, **kwargs):
+        """Check if community can access restricted files of the migrated record."""
+        try:
+            can_community_read_files = dict_lookup(
+                record.parent, "permission_flags.can_community_read_files"
+            )
+        except KeyError:
+            can_community_read_files = True
+
+        is_restricted = super()._condition(record, **kwargs)
+        if is_restricted:
+            return not can_community_read_files
```

Comment:
I'm not quite sure I understand why the `not` is needed. Thinking of the following case:

- The record has the `can_community_read_files` flag set to `False`. Then `can_community_read_files=False`.
- The files are restricted, therefore the code will enter the `if` in line 31. And will return `not can_community_read_files`, which is `not False`, therefore returning `True`. 

I might have misunderstood something on the flow, but I think is wrong and could lead to data leak. Shouldn't the return's be like:

```python
return can_community_read_files if restricted else False
```

Meaning: if files are restricted return what the flag has as value, otherwise default to false

Reply:
In case of `can_community_read_files = False`, the condition of class `IfFilesRestrictedForCommunity` will return true, which means that files are restricted for community (as we can understand from the name of the class).

So in case of the returning value being `True`, the permission `can_read_files_owner` will be used. So in the end if the files are restrcited and `can_community_read_files = False` will lead to the permission logic where community can not see files.

Probably the naming is confusing, but for the better understanding you can check the tests (hope it helps).